pkg/test/framework/components/istio/kube.go

```diff
@@ -214,16 +214,16 @@ func (i *istioImpl) RemoteDiscoveryAddressFor(cluster cluster.Cluster) (netip.Ad
 		// istiod is exposed via LoadBalancer since we won't have ingress outside of a cluster;a cluster that is;
 		// a control cluster, but not config cluster is supposed to simulate istiod outside of k8s or "external"
 		address, err := retry.UntilComplete(func() (any, bool, error) {
-			return getRemoteServiceAddress(i.env.Settings(), primary, i.cfg.SystemNamespace, istiodLabel,
-				istiodSvcName, discoveryPort)
+			addrs, outcome, err := getRemoteServiceAddresses(i.env.Settings(), primary, i.cfg.SystemNamespace, istiodLabel, istiodSvcName, discoveryPort)
+			return addrs[0], outcome, err
```

Comment:
Is this func you mentioned as "requiring extensive changes" if modified to return multiple addresses ?

Reply:
DiscoveryAddresses is used in two functions:
- `getVMOverrideForIstiodDNS` which is used as part of a workaround when deploying echo in VMs, this can be easily adapted to be DS by tinkering with echo vm_deployment.
- `RemoteDiscoveryAddressFor` which is in turn used in: creating the VM config for echo deployments, input in `values.global.remotePilotAddress `and `ISTIOCTL_XDS_ADDRESS` env variable. All of these do not account for the possibility of multiple addresses and I don't think there is a usecase to justify the changes since I don't expect the user to provide two addresses at the same time. Therefore my approach would have been to rerun all the affected tests twice as the setup would need to change, which I feel would end up having big impact